flink-runtime/src/test/java/org/apache/flink/runtime/operators/chaining/ChainedOperatorsMetricTest.java

```diff
@@ -68,22 +70,26 @@
     public void testOperatorIOMetricReuse() throws Exception {
         // environment
         initEnvironment(MEMORY_MANAGER_SIZE, NETWORK_BUFFER_SIZE);
+
         this.mockEnv =
                 new MockEnvironmentBuilder()
                         .setTaskName(HEAD_OPERATOR_NAME)
                         .setManagedMemorySize(MEMORY_MANAGER_SIZE)
                         .setInputSplitProvider(this.inputSplitProvider)
                         .setBufferSize(NETWORK_BUFFER_SIZE)
                         .setMetricGroup(
-                                new TaskMetricGroup(
-                                        NoOpMetricRegistry.INSTANCE,
-                                        UnregisteredMetricGroups
-                                                .createUnregisteredTaskManagerJobMetricGroup(),
-                                        new JobVertexID(),
-                                        new ExecutionAttemptID(),
-                                        "task",
-                                        0,
-                                        0))
+                                new TaskManagerJobMetricGroup(
+                                                NoOpMetricRegistry.INSTANCE,
+                                                UnregisteredMetricGroups
+                                                        .createUnregisteredTaskManagerMetricGroup(),
+                                                new JobID(0, 0),
+                                                "UnregisteredJob")
+                                        .addTask(
+                                                new JobVertexID(),
+                                                new ExecutionAttemptID(),
+                                                "task",
+                                                0,
+                                                0))
```

Comment:
```suggestion
                                MetricUtils.createTaskManagerMetricGroup(
                                                NoOpMetricRegistry.INSTANCE,
                                                "host",
                                                ResourceID.generate())
                                        .addTaskForJob(
                                                new JobID(),
                                                "jobname",
                                                new JobVertexID(),
                                                new ExecutionAttemptID(),
                                                "task",
                                                0,
                                                0))
```
lets do it like this.

Reply:
Thanks for your direction